File: FbxSharpTests/MeshTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class MeshTest : TestBase
    {
        [Test]
        public void Mesh_Create()
        {
            // given:
            var mesh = new FbxMesh("Mesh");

            // then:
            Assert.AreEqual(0, mesh.GetControlPointsCount());
        }

        [Test]
        public void Mesh_Create_HasProperties()
        {
            // given:
            var mesh = new FbxMesh("Mesh");
            FbxProperty prop;

            // then:
            Assert.AreEqual(6, CountProperties(mesh));
            Assert.AreEqual(0, mesh.GetSrcPropertyCount());
            Assert.AreEqual(0, mesh.GetDstPropertyCount());

            prop = mesh.FindProperty("Color");
            Assert.NotNull(prop);
            Assert.True(prop.IsValid());
            Assert.NotNull(mesh.Color);
            Assert.True(mesh.Color.IsValid());
            Assert.AreEqual("Color", mesh.Color.GetName());
            Assert.AreSame(prop, mesh.Color);

            prop = mesh.FindProperty("Primary Visibility");
            Assert.NotNull(prop);
            Assert.True(prop.IsValid());
            Assert.NotNull(mesh.PrimaryVisibility);
            Assert.True(mesh.PrimaryVisibility.IsValid());
            Assert.AreEqual("Primary Visibility", mesh.PrimaryVisibility.GetName());
            Assert.AreSame(prop, mesh.PrimaryVisibility);

            prop = mesh.FindProperty("Casts Shadows");
            Assert.NotNull(prop);
            Assert.True(prop.IsValid());
            Assert.NotNull(mesh.CastShadow);
            Assert.True(mesh.CastShadow.IsValid());
            Assert.AreEqual("Casts Shadows", mesh.CastShadow.GetName());
            Assert.AreSame(prop, mesh.CastShadow);

            prop = mesh.FindProperty("Receive Shadows");
            Assert.NotNull(prop);
            Assert.True(prop.IsValid());
            Assert.NotNull(mesh.ReceiveShadow);
            Assert.True(mesh.ReceiveShadow.IsValid());
            Assert.AreEqual("Receive Shadows", mesh.ReceiveShadow.GetName());
            Assert.AreSame(prop, mesh.ReceiveShadow);

            prop = mesh.FindProperty("BBoxMin");
            Assert.NotNull(prop);
            Assert.True(prop.IsValid());
            Assert.NotNull(mesh.BBoxMin);
            Assert.True(mesh.BBoxMin.IsValid());
            Assert.AreEqual("BBoxMin", mesh.BBoxMin.GetName());
            Assert.AreSame(prop, mesh.BBoxMin);

            prop = mesh.FindProperty("BBoxMax");
            Assert.NotNull(prop);
            Assert.True(prop.IsValid());
            Assert.NotNull(mesh.BBoxMax);
            Assert.True(mesh.BBoxMax.IsValid());
            Assert.AreEqual("BBoxMax", mesh.BBoxMax.GetName());
            Assert.AreSame(prop, mesh.BBoxMax);
        }

        [Test]
        public void Mesh_Create_HasNamespacePrefix()
        {
            // given:
            var obj = new FbxMesh("asdf");

            // then:
            Assert.AreEqual("Geometry::", obj.GetNameSpacePrefix());
        }
    }
}

File: FbxCppTests/MeshTest.cpp

#include "Tests.h"

using namespace std;

void Mesh_Create()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxMesh* mesh = FbxMesh::Create(manager, "Mesh");

    // then:
    AssertEqual(0, mesh->GetControlPointsCount());
}

void Mesh_Create_HasProperties()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxMesh* mesh = FbxMesh::Create(manager, "Mesh");
    FbxProperty prop;

    // then:
    AssertEqual(6, CountProperties(mesh));
    AssertEqual(0, mesh->GetSrcPropertyCount());
    AssertEqual(0, mesh->GetDstPropertyCount());

    prop = mesh->FindProperty("Color");
    AssertTrue(prop.IsValid());
    AssertTrue(mesh->Color.IsValid());
    AssertEqual("Color", mesh->Color.GetName());

    prop = mesh->FindProperty("Primary Visibility");
    AssertTrue(prop.IsValid());
    AssertTrue(mesh->PrimaryVisibility.IsValid());
    AssertEqual("Primary Visibility", mesh->PrimaryVisibility.GetName());

    prop = mesh->FindProperty("Casts Shadows");
    AssertTrue(prop.IsValid());
    AssertTrue(mesh->CastShadow.IsValid());
    AssertEqual("Casts Shadows", mesh->CastShadow.GetName());

    prop = mesh->FindProperty("Receive Shadows");
    AssertTrue(prop.IsValid());
    AssertTrue(mesh->ReceiveShadow.IsValid());
    AssertEqual("Receive Shadows", mesh->ReceiveShadow.GetName());

    prop = mesh->FindProperty("BBoxMin");
    AssertTrue(prop.IsValid());
    AssertTrue(mesh->BBoxMin.IsValid());
    AssertEqual("BBoxMin", mesh->BBoxMin.GetName());

    prop = mesh->FindProperty("BBoxMax");
    AssertTrue(prop.IsValid());
    AssertTrue(mesh->BBoxMax.IsValid());
    AssertEqual("BBoxMax", mesh->BBoxMax.GetName());
}

void Mesh_Create_HasNamespacePrefix()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxMesh* obj = FbxMesh::Create(manager, "asdf");

    // then:
    AssertEqual("Geometry::", obj->GetNameSpacePrefix());
}

void MeshTest::RegisterTestCases()
{
    AddTestCase(Mesh_Create);
    AddTestCase(Mesh_Create_HasProperties);
    AddTestCase(Mesh_Create_HasNamespacePrefix);
}

